examples/Ballot.bl

```diff
@@ -0,0 +1,89 @@
+version 1.1
+
+contract Ballot {
+
+    struct Voter {
+        int weight
+        bool voted
+        address delegate
+        int vote
+    }
+
+    struct Proposal {
+        String name
+        int voteCount
+    }
+
+    address chairperson
+
+    Map<address, Voter> voters
+
+    Proposal[] proposals
+
+    constructor(String[] proposalNames) {
+        chairperson = msg.sender
+        voters[chairperson].weight = 1
+        proposals = new Proposal[proposalNames.length]
+
+        foreach(i, String name : proposalNames) {
+            proposals[i] = new Proposal(name, 0)
+        }
+    }
+
+    [Pre=msg.sender == chairperson]
+    [Pre=!voters[voter].voted]
+    [Pre=voters[voter].weight == 0]
+    function void giveRightToVote(address voter) {
+        voters[voter].weight = 1
+    }
+
+    [Pre = !voters[msg.sender].voted]
```

Comment:
This colud look confusing, one might want to write Pre != voters... What about Pre: